test-utils/tests/dir.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2022 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[test]
+fn existing_dir_error() {
+    pub const TARGET_TMPDIR: &str = env!("CARGO_TARGET_TMPDIR");
+    // Target tmpdir should have been created by cargo, check we get the expected error code
+    match std::fs::create_dir(TARGET_TMPDIR) {
```

Comment:
I wonder whether we can wrap this properly so that we're testing our own function instead of testing rust's behavior.

Reply:
The intention is to test Rust's behaviour here, just wanted to make sure what is the error we get when creating an existing dir. I will think a bit how to test our function in addition to it but I'd like to keep this around.

Reply:
I completely understand what you're trying to achieve. But instead of testing what a generic function does, just wrap it with your own call with proper constraints. For example `create_new_dir_or_panic()`. This way, if someone else sees the logic and thinks of changing it with something else in the future, they realize that it's an invariant in this test that a panic should happen if the dir already exists. Feel free to keep this, but it just seems more logical to restrict behavior to our code.

Reply:
Created a simple wrapper that is used in both our tests and in the code returning an enum to distinguish between already existing and created dir. Maybe a bit overkill but worked out quite well in the end. It's in the last commit, please take a look.

Reply:
Looking good. Thank you!